File: ocaml/src/five.ml
open OUnit2

let rev =
  let rec go reversed =
    function | [] -> reversed | x::rest -> go (x :: reversed) rest in
  go []

let test_case =
  "rev" >:: (fun _  -> assert_equal ["c"; "b"; "a"] (rev ["a"; "b"; "c"]))

File: fsharp/Five.fs
module ``NinetyNine Problems - Problem 5``

open NUnit.Framework
open FsUnit

let rev =
  let rec aux reversed = function
    | x :: rest -> aux (x :: reversed) rest
    | [] -> reversed
  in
    aux []

[<Test>]
let ``Given a list it should return its reversed version`` () =
  rev [ "a" ; "b"; "c"; "d"; "e" ] |> should equal [ "e" ; "d" ; "c" ; "b" ; "a" ]
